### PR TITLE
[fix issue and improvement] verify_no_pre_exist_users

### PR DESCRIPTION
**Pull request by LiliDeng**

### CODE REVIEW

File: microsoft/testsuites/core/azure_image_standard.py [hunk 1]
```diff
@@ -892,22 +893,43 @@ def verify_no_pre_exist_users(self, node: Node) -> None:
...
+            if shadow_raw_output.strip().startswith("#"):
                 continue
             # sample line of /etc/shadow
             # root:x:0:0:root:/root:/bin/bash
             # sshd:!:19161::::::
             # systemd-coredump:!*:19178::::::
             # get first two columns of /etc/shadow
-            user_name, user_passwd = passwd_raw_output.split(":")[0:2]
-            if not ("*" in user_passwd or "!" in user_passwd or "x" == user_passwd):
+            user_name, user_passwd = shadow_raw_output.split(":")[0:2]
```
Comment (squirrelsc) [inline microsoft/testsuites/core/azure_image_standard.py]: change to regular expressions.

Comment (LiliDeng) [reply microsoft/testsuites/core/azure_image_standard.py]: updated

File: microsoft/testsuites/core/azure_image_standard.py [hunk 2]
```diff
@@ -881,33 +881,71 @@ def verify_client_active_interval(self, node: Node) -> None:
...
+                continue
+            # sample line of /etc/passwd
+            # centos:x:1000:1000::/home/centos:/bin/bash
+            passwd_matches = pattern.findall(passwd_raw_output)
+            assert passwd_matches[5], "not found home folder column of file /etc/passwd"
+            file_path = f"{passwd_matches[5]}/.ssh/authorized_keys"
+            file_exists = node.tools[Ls].path_exists(file_path, sudo=True)
+            if file_exists:
+                key_content = node.tools[Cat].read(file_path, sudo=True)
+                key_match = key_pattern.findall(key_content)
```
Comment (squirrelsc) [inline microsoft/testsuites/core/azure_image_standard.py:943]: This logic is not easy to understand, and some comment here.

Comment (LiliDeng) [reply microsoft/testsuites/core/azure_image_standard.py:943]: updated

**Review by squirrelsc — DISMISSED** [review]

**Pull request merged**